### ISSUE TITLE
Incompatible value passed to lru memory cache set function

### ISSUE THREAD

**Issue by edcjrh** [node-cache-manager/node-cache-manager#280]
Hi,

When I am using multicache (Redis +Mmeory) the ttl passed in the .set function is usually in this form:

```javascript
await multiCache.set(id, document, {ttl});
```

This works fine for the Redis cache store but it does not work for the memory store that it is based on LRU.

When the ttl value is passed to the memory store set function then it is received as an object and passed as it is:

```javascript
return __generator(this, function (_a) {
                    if (!isCacheable(value))
                        throw new Error("no cacheable value ".concat(JSON.stringify(value)));
                    if (shouldCloneBeforeSet)
                        value = clone(value);
                    ttl = opt ? opt : lruOpts.ttl;
                    lruCache.set(key, value, { ttl: ttl });
                    return [2 /*return*/];
                });
```

Then the actual value passed to lruCache will be {ttl: {ttl: <number>}}, this will cause the memory cache to not expire.

Am I missing something?

**Comment by botika** [comment]
https://github.com/node-cache-manager/node-cache-manager/blob/675a8ad53b8207efa37099606f0fb0ba7c809c29/README.md?plain=1#L90-L93

**Comment by edcjrh** [comment]
If I do that then the Redis cache won't work. I have tested both and that it is why I end up with a workaround where I pass it for mem cahe as you show but for redis it needs to be {ttl}

**Comment by botika** [comment]
You have to check that the version of Redis-store is correct.

**Comment by edcjrh** [comment]
Hi,

Thanks for re-opening the issue. :-)

I have checked and updated because the new redis@4. Here are the versions I am using:

```json 
"cache-manager": "5.1.3",

"cache-manager-redis-store": "3.0.1",
```

So I did this test:
store[0] is the mem store and store[1] is the redis store. By doing this the ttl is honored and both cache stores will remove the values when the ttl expires.

```javascript
            // This will ensure that the values passed are correct for both memory and Redis
            await this.stores[0].store.set(id, document, ttl * 1000);
            await this.stores[1].store.set(id, document, {ttl});
```

If I don't use {ttl} in the Redis store, but just ttl then it will use the default value given when is created.

And in the memory store case if I don't give the ttl * 1000 (milliseconds), then it will take the value in milliseconds, but the default value at creation works fine.


```javascript
const memoryOptions = {
            max: 100,
            ttl: 180 # 3 minutes for cache in memory
        };
const memoryCache = await cacheManager.caching("memory", this.memoryOptions);
```
For this case if I don't pass a ttl to the set function the cache will use the one if the definition, but if I pass the ttl as {ttl} it will give an undefined value for LRU causing to never remove the data from the cache.

Not sure if the explanation is good enough

**Comment by botika** [comment]
Use [cache-manager-redis-yet](https://github.com/node-cache-manager/node-cache-manager-redis-yet) or [cache-manager-ioredis-yet](https://github.com/node-cache-manager/node-cache-manager-ioredis-yet) at last versions

**Comment by edcjrh** [comment]
I will give it a try, thanks

**Comment by edcjrh** [comment]
I checked the tests run for node-cache-manager and I noticed that you use the mem store for all. 
The problem I have seen is the Redis store receives the ttl in seconds, while the mem store receives the store in milliseconds.
That causes the mem store to remove the values before the redis store has removed them.
If I pass ttl = 5 to multicaching:

```javascript
multiCache = multiCaching([memoryCache, RedisCache]);
multiCache.set(key, value, 5);
```

For the redis store this value is given in seconds, but for the mem cache the value is in milliseconds.

So the times when the data will expire are different.
